packages/requests/src/lib/requests-result.ts

```diff
@@ -26,12 +28,13 @@ export interface LoadingRequestResult extends BaseRequestResult {
   fetchStatus: 'fetching' | 'idle';
 }
 
-export interface SuccessRequestResult extends BaseRequestResult {
+export interface SuccessRequestResult<TData = any> extends BaseRequestResult {
   isLoading: false;
   isSuccess: true;
   isError: false;
   status: 'success';
   fetchStatus: 'idle';
+  responseData?: TData;
```

Comment:
Note that when reading this value it will always be TData or undefined even if guard is use